src/precice/impl/DataContext.cpp

```diff
@@ -29,6 +29,7 @@ void DataContext::resetData()
     PRECICE_ASSERT(!hasReadMapping());
     std::for_each(_mappingContexts.begin(), _mappingContexts.end(), [](auto &context) { context.toData->toZero(); });
   }
+  // @todo need to also reset/clear (write)DataBuffer?
```

Comment:
No. `_providedData->toZero()` only resets the `Data::_sample`, but the write data buffer only exists in the `WriteDataContext`. Therefore, we would need to somehow solve this via inheritance, if we want to also reset `WriteDataContext::_writeDataBuffer`. I guess it does not harm and since we want to get rid of `Data::_sample` on the long run, we will need it anyway.

Reply:
Actually `resetData` is only needed for `WriteDataContext`. This simplifies things. 8748fd773c6ed597d7b618c37b2eb8e2978e3063